### PR TITLE
PG wire - bound parameters in nodejs fix #290

### LINKED ISSUE #290: Bound parameters not working when using PostgreSQL wire protocol through nodejs
How to reproduce:

`pg-client.js`
```javascript
const { Client } = require("pg")

const start = async () => {
  try {
    const client = new Client({
      database: "qdb",
      host: "127.0.0.1",
      password: "quest",
      port: 8812,
      user: "admin",
    })
    await client.connect()

    const res = await client.query(
      "SELECT * FROM test WHERE id = $1",
      ["1"],
    )

    console.log(res.rows[0])

    await client.end()
  } catch (e) {
    console.log(e)
  }
}

start()
```

`package.json`
```json
{
  "name": "example",
  "version": "1.0.0",
  "license": "Apache-2.0",
  "dependencies": {
    "pg": "^8.2.0"
  }
}
```

Then run:

- `npm i` to install the dependencies
- `node pg-client.js` to run the code

You should get the following error:
```
error: no bind variable defined at index 0
```

### PR DESCRIPTION
**Pull request by jaugsburger**

### CODE REVIEW

File: core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java [hunk 1]
```diff
@@ -518,6 +540,12 @@ private void appendByteColumn(Record record, int columnIndex) {
         responseAsciiSink.putLenEx(a);
     }
 
+    private void appendCharColumn(Record record, int columnIndex) {
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java:543]: This method is introduced but not tested. How does this map to PG type?

File: core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java [hunk 2]
```diff
@@ -690,11 +702,50 @@ private void bindVariables(
                 throw BadProtocolException.INSTANCE;
             }
             ensureData(lo, valueLen, msgLimit, j);
+            //infer type if needed
+            if (inferTypes) {
+                int pgType = inferParameterType(lo, valueLen);
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java:707]: `inferParameterType()` is liable to return -1, which is not handled and not tested either.

File: core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java [hunk 3]
```diff
@@ -817,6 +868,16 @@ private CharSequence getStatementName(long lo, long hi) throws BadProtocolExcept
...
     }
 
+    private int inferParameterType(long lo, int valueLen) {
+        for (int i = 0; i < typeManager.getProbeCount(); i++) {
+            TypeAdapter typeAdapter = typeManager.getProbe(i);
+            if (typeAdapter.probe(parameterHolder.of(lo, lo + valueLen))) {
+                return typeOids.get(typeAdapter.getType());
+            }
+        }
+        return -1;
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java:878]: tests don't get here

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java (hunk 1 above)
Comment (jaugsburger) [reply core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java]: sorry, removed as not used

File: core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java (hunk 2 above)
Comment (jaugsburger) [reply core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java:701]: fixed

File: core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java (hunk 3 above)
Comment (jaugsburger) [reply core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java:876]: added test

**Review by bluestreak01 — APPROVED** [review]
we will need to improve test coverage

**Pull request merged**